STUDENT_COURSE _MANAGEMENT/src/main/java/com/springboot/demo/studentcourse/config/DemoSecurityConfig.java

```diff
@@ -0,0 +1,51 @@
+package com.springboot.demo.studentcourse.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+import javax.sql.DataSource;
+
+@Configuration
+@EnableWebSecurity
+public class DemoSecurityConfig extends WebSecurityConfigurerAdapter {
+
+
+    @Autowired
+    private DataSource securityDataSource;
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+
+
+        auth.jdbcAuthentication().dataSource(securityDataSource);
+
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+
+        http.authorizeRequests()
+                .antMatchers("/students").hasAnyRole("ADMIN","FACULTY")
+                .antMatchers("/courses/**").hasRole("ADMIN")
+                .antMatchers("/students/showFormForAdd").hasRole("ADMIN")
```

Comment:
can't we use students/** for ADMIN because admin has all authority right!

Reply:
We can but it's giving error so I seperated them

Reply:
what is that error?

Reply:
Access denied page